tests/Tests/100__deploy/100__installation/102__post_install.robot

```diff
@@ -239,3 +251,63 @@ Verify Errors In Jupyterhub Logs
+    ...    Args:
+    ...        project: Project name
+    ...    Returns:
+    ...        None
+    [Arguments]    ${project}
+    ${project_pods_info}=    Fetch Project Pods Info    ${project}
+    FOR    ${pod_info}    IN    @{project_pods_info}
+        Verify CPU And Memory Requests And Limits Are Defined For Pod    ${pod_info}
+    END
+
+Verify CPU And Memory Requests And Limits Are Defined For Pod
+    [Documentation]    Verifies that CPU and memory requests and limits are defined
+    ...                for the specified pod
+    ...    Args:
+    ...        pod_info: Pod information
+    ...    Returns:
+    ...        None
+    [Arguments]    ${pod_info}
+    &{pod_info_dict}=    Set Variable    ${pod_info}
+    FOR    ${container_info}    IN    @{pod_info_dict.spec.containers}
+        Verify CPU And Memory Requests And Limits Are Defined For Pod Container    ${container_info}
+    END
+
+Verify CPU And Memory Requests And Limits Are Defined For Pod Container
+    [Documentation]    Verifies that CPU and memory requests and limits are defined
+    ...                for the specified container
+    ...    Args:
+    ...        container_info: Container information
+    ...    Returns:
+    ...        None
+    [Arguments]    ${container_info}
+    &{container_info_dict} =    Set Variable    ${container_info}
+    OpenShift Resource Component Should Contain Field     ${container_info_dict}    resources
+    Run Keyword If   'resources' in ${container_info_dict}
+    ...    OpenShift Resource Component Should Contain Field     ${container_info_dict.resources}    requests
+    Run Keyword If   'resources' in ${container_info_dict}
+    ...    OpenShift Resource Component Should Contain Field     ${container_info_dict.resources}    limits
+    Run Keyword If   'requests' in ${container_info_dict.resources}
+    ...    OpenShift Resource Component Should Contain Field     ${container_info_dict.resources.requests}    cpu
+    Run Keyword If   'requests' in ${container_info_dict.resources}
+    ...    OpenShift Resource Component Should Contain Field     ${container_info_dict.resources.requests}    memory
+    Run Keyword If   'limits' in ${container_info_dict.resources}
+    ...    OpenShift Resource Component Should Contain Field     ${container_info_dict.resources.limits}    cpu
+    Run Keyword If   'limits' in ${container_info_dict.resources}
+    ...    OpenShift Resource Component Should Contain Field     ${container_info_dict.resources.limits}    memory
+
+Fetch Project Pods Info
+    [Documentation]    Fetches information of all Pods for the specified Project
+    ...    Args:
+    ...        project: Project name
+    ...    Returns:
+    ...        project_pods_info: List of Project Pods information
+    [Arguments]    ${project}
+    @{project_pods_info}=    Oc Get    kind=Pod    api_version=v1    namespace=${project}
+    [Return]    @{project_pods_info}
```

Comment:
I'm wondering if these three keywords should be defined in a resource file? They look general enough to be reused in other contexts to be honest

Reply:
+1

Reply:
done